### PR TITLE
A rudimentary scripting interface for i-PI

### PR DESCRIPTION
**Pull request by ceriottm**
This PR implements a rudimentary scripting API, mostly by just restructuring the Simulation class and providing a few utilities to build a simulation on the fly. Will give the best user experience after #390 is merged

### CODE REVIEW

**Comment by ceriottm** [comment]
This is ready for a first review. @venkatkapil24 please take care of the ASE units.
I will also try and add a short section to the documentation mentioning that there is now this possibility.
I think probably in the long run we should think of transitioning to sphinx-generated documentation of the API, but this is enough for a first trial IMO.

File: docs/src/python.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        forcefield=forcefield_xml(name="dummy", mode="direct"),
+        motion=motion_nvt_xml(timestep=2.0 * ase.units.fs),
+        temperature=300,
+        prefix="example",
+    )
+
+    sim = InteractiveSimulation(input_xml)
+    sim.run(100)
+    potential = sim.properties("potential")
+    ase.io.write("final_structure.xyz", sim.get_structures())
```
[the remark this answers is not in the document]
Comment (ceriottm) [reply docs/src/python.rst:64]: done

**Comment by venkatkapil24** [comment]
@ceriottm are you sure the branch is up to date with your local version? I can't run the metatensor example to understand what the 'ASE units issue' is.

**Comment by venkatkapil24** [comment]
FYI, here is the error I was getting. 

 File "/Users/venkatkapil24/source/i-pi/ipi/utils/scripting.py", line 93
    f"""<initialize nbeads='{nbeads}'>
        ^
SyntaxError: f-string: expecting '}

It's fixed now. I wonder if this is a Python version-based error, if it worked on @ceriottm's setup.

File: ipi/pes/ase.py [hunk 2]
```diff
@@ -21,16 +21,34 @@ class ASEDriver(Dummy_driver):
...
+        :param has_energy: bool, whether the model can compute energy
+        :param has_forces: bool, whether the model can compute forces
+        :param has_stress: bool, whether the model can compute stress
     """
 
-    def __init__(self, template, *args, **kwargs):
+    def __init__(
+        self,
+        template,
+        has_energy=True,
```
[the remark this answers is not in the document]
Comment (ceriottm) [reply ipi/pes/ase.py:32]: I think so, most potentials will provide all of these so we shouldn't force ppl to add these on the command line.

**Review by mahrossi — APPROVED** [review]

**Pull request merged**